dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/bundle/TrackerSideEffectHandlerServiceTest.java

```diff
@@ -129,6 +140,20 @@ public void testRuleEngineSideEffectHandlerService()
 
         TrackerBundleReport report = trackerBundleService.commit( trackerBundles.get( 0 ) );
 
+        List<TrackerSideEffectDataBundle> sideEffectDataBundles = ListUtils.union(
+            report.getTypeReportMap().get( TrackerType.ENROLLMENT ).getSideEffectDataBundles(),
+            report.getTypeReportMap().get( TrackerType.EVENT ).getSideEffectDataBundles() );
+
+        trackerBundleService.handleTrackerSideEffects( sideEffectDataBundles );
+
         assertEquals( report.getStatus(), TrackerStatus.OK );
+
+        await().atMost( 10, TimeUnit.SECONDS ).until( () -> manager.getAll( ProgramNotificationInstance.class ).size() > 0 );
```

Comment:
10 seconds is a bit too long for a test? Have you considered lowering to 1 or 2 secs?

Reply:
set it to 2

Reply:
2 also sounds a bit high :-) Do we need this? I.e. is there a way to mock or handle it differently?

Reply:
Side effects are handled asynchronously. So this is one of the ways to test thread functionality.